### PR TITLE
Enable GitOps and Canaries in Development and Integration Environment

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description
Enable feature flag and migrate resource requests/limits

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

----

 - [ ] Check quay-ip secret, is it created?
 - [ ] Check if environment variables / image pull secrets are available in openshift-ci?
 - [ ] Populate and link secret

### CODE REVIEW

File: .openshift-ci/tests/e2e.sh [hunk 1]
```diff
@@ -9,8 +9,10 @@ export GITROOT
 # shellcheck source=/dev/null
 source "${GITROOT}/dev/env/scripts/lib.sh"
 
-RUN_AUTH_E2E_DEFAULT="false"
-RUN_CENTRAL_E2E_DEFAULT="true"
+export RUN_AUTH_E2E="false"
+export RUN_CENTRAL_E2E="true"
+export RHACS_STANDALONE_MODE="true"
+export RHACS_GITOPS_ENABLED="true"
```
Comment (kurlov) [inline .openshift-ci/tests/e2e.sh:15]: We should drop it for now

File: dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml [hunk 2]
```diff
@@ -67,6 +67,10 @@ spec:
                 secretKeyRef:
                   name: fleetshard-sync
                   key: "aws-secret-access-key"
+            - name: RHACS_GITOPS_ENABLED
```
Comment (kurlov) [inline dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml:70]: It's duplicated

**Review by ludydoo — APPROVED** [review]

File: e2e/e2e_canary_upgrade_test.go [hunk 3]
```diff
@@ -140,7 +143,7 @@ var _ = Describe("Fleetshard-sync Targeted Upgrade", func() {
 			gitops.RHACSOperators.Configs = []operator.OperatorConfig{operatorConfig1}
 			err = updateGitopsConfig(ctx, gitops)
 			Expect(err).To(BeNil())
-			Eventually(expectNumberOfOperatorDeployments(ctx, 1)).
+			Eventually(expectNumberOfOperatorDeployments(ctx, 1, "")).
```
Comment (ludydoo) [inline e2e/e2e_canary_upgrade_test.go:146]: `""` ?

Comment (SimonBaeumer) [reply e2e/e2e_canary_upgrade_test.go]: done

File: fleetshard/pkg/central/operator/upgrade.go [hunk 4]
```diff
@@ -110,18 +112,27 @@ func (u *ACSOperatorManager) RemoveUnusedOperators(ctx context.Context, desiredI
 
 	var unusedDeployments []string
 	for _, deployment := range deployments.Items {
-		for _, container := range deployment.Spec.Template.Spec.Containers {
-			if container.Name == "manager" && !slices.Contains(desiredImages, container.Image) {
-				unusedDeployments = append(unusedDeployments, deployment.Name)
+		found := false
+		for _, expectedOperator := range operatorConfigs {
+			if expectedOperator.GetDeploymentName() == deployment.GetName() {
+				found = true
```
Comment (ludydoo) [inline fleetshard/pkg/central/operator/upgrade.go:116]: `
found = true
break
`

**Pull request closed without merging**